packet/models_tinkerbell.go

```diff
@@ -89,7 +89,10 @@ func (d *DiscoveryTinkerbellV1) PrimaryDataMAC() MACAddr {
 }
 
 func (d DiscoveryTinkerbellV1) Hostname() (string, error) {
-	return d.Instance().Hostname, nil // temp
+	if d.Instance() == nil {
+		return "", nil
```

Comment:
There is some interdependency of `*DiscoveryTinkerbellV1` struct with `Discovery` struct. I tried removing it but the test run failed. As of now, I thought it would be better to keep this same as in an existing way and modify later.

I am thinking to refactor the code at some point. There are many similar methods which need error handling.